### PR TITLE
[Fix] HttpClient concurrent usage.

### PR DESCRIPTION
**Pull request by ruseinov**
## Summary of changes

Changes introduced in this pull request:

- Hyper http client is only built once and reused via shallow clones as it should be.

## Reference issue to close (if applicable)

Fixes https://github.com/ChainSafe/forest/issues/2967

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: utils/forest_utils/src/net/http.rs [hunk 1]
```diff
@@ -4,17 +4,25 @@
...
+use lazy_static::lazy_static;
 
-/// Constructs [hyper::Client] that supports both `http` and `https`.
-/// Note that only `http1` is supported.
-pub fn https_client() -> hyper::Client<HttpsConnector<HttpConnector>> {
-    hyper::Client::builder().build::<_, Body>(
+lazy_static! {
+    /// A default [hyper::Client]. It's imperative that the builder is only
+    /// called once, because fetching root certificates is expensive.
+    static ref CLIENT: hyper::Client<HttpsConnector<HttpConnector>> =
```
Comment (LesnyRumcajs) [inline utils/forest_utils/src/net/http.rs:12]: Perhaps a `OnceCell` would be a bit cleaner. Not a huge deal, though.

Comment (ruseinov) [reply utils/forest_utils/src/net/http.rs:12]: > Perhaps a `OnceCell` would be a bit cleaner. Not a huge deal, though.

Totally, but we are not on the 1.70 toolchain yet as far as I know.

Comment (ruseinov) [reply utils/forest_utils/src/net/http.rs:12]: I think perhaps `Lazy` is a better option here, I'll incorporate that after raising a PR to bump the version.

**Comment by aatifsyed** [comment]
I'd like to see our client deprecated and replaced with `reqwest`.
- We don't need to (re)-implement standard logic for redirects and well-known headers - see https://github.com/ChainSafe/forest/blob/5532ab90cd69f3dce313501d29d99c5ae99a0de6/utils/forest_utils/src/net/download.rs#L181, https://github.com/ChainSafe/forest/blob/5532ab90cd69f3dce313501d29d99c5ae99a0de6/utils/forest_utils/src/proofs_api/paramfetch.rs#L195, and https://github.com/ChainSafe/forest/blob/5532ab90cd69f3dce313501d29d99c5ae99a0de6/utils/forest_utils/src/net/download.rs#L170

**Comment by ruseinov** [comment]
> I'd like to see our client deprecated and replaced with `reqwest`.

Totally agreed!

It'd be nice to have an issue for that and do it in a dedicated PR. This PR aims at fixing something that totally breaks node startup on some machines, including mine and is a pretty easy and fast one to review and merge, therefore I'd like to keep it's scope as small as possible.

**Comment by ruseinov** [comment]
@aatifsyed https://github.com/ChainSafe/forest/issues/2977

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**